doc/userguide/rules/smb-keywords.rst

```diff
@@ -0,0 +1,16 @@
+SMB Keywords
+==============
+
+SMB keywords used in both SMB1 and SMB2 protocols.
+
+smb_filename
+--------------
+
+SMB filename is an sticky buffer to match the filename in SMB Create requests.
+
+.. note:: Remember that SMB2 filenames are Unicode encoded.
+
+If you want to match traffic that access to file "a.txt", you could use the following rule::
+
+  alert smb any any -> any any (msg: "SMB file match";smb.filename; content:"a|00|.|00|t|00|x|00|t|00|";sid:1;)
```

Comment:
In our logger we log with `String::from_utf8_lossy`. Wonder if we should either (1) add a transform keyword like that, or (2) default this detection to do this.

So you could either have:
`alert smb any any -> any any (msg: "SMB file match";smb.filename; content:"a.txt";sid:1;)`
or
`alert smb any any -> any any (msg: "SMB file match";smb.filename; from_utf8_lossy; content:"a.txt";sid:1;)`

Reply:
I'm not sure if default makes sense here? I'm not sure about in the Windows/SMB world, but in unix-land filenames do not have to be value UTF-8, so defaulting to `from_utf8_loss` would make exact matches in some case impossible, if that matters.

Reply:
In Windows, filenames can have almost any Unicode character (https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#naming-conventions).

It would be great to being able match unicode strings (quite useful for Windows world). A possible way could be specify an u before the string, like for example u"a.txt" to indicate the match of unicode. The rule would be this way: 

`alert smb any any -> any any (msg: "SMB file match";smb.filename; content:u"a.txt";sid:1;)`